### PR TITLE
feat(permissions): NASS-1906: Permissions matrix admin frontend

### PR DESCRIPTION
**Pull request by chris-bes**
### Changes
- Permissions matrix admin frontend

### Deploys

- [x] **Deploy to Tamanu Internal** 

### Tests

- [ ] **Run E2E Tests** 

### Review Hero

- [ ] **Run Review Hero** 
- [ ] **Auto-fix review suggestions**  _Wait for Review Hero to finish, resolve any comments you disagree with or want to fix manually, then check this to auto-fix the rest._
- [ ] **Auto-fix CI failures**  _Check this to auto-fix lint errors, test failures, and other CI issues._

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

---

> [!NOTE]
> **Medium Risk**
> Adds new admin UI and mutations that can create/delete permissions in bulk, and changes server-side permission matrix generation/validation; mistakes could grant/strip access across roles. Backend changes are scoped but touch authorization data paths.
> 
> **Overview**
> Adds an **Edit** tab to the admin Permissions page that renders a new permissions matrix UI with role and noun/objectId filtering, expandable noun rows, and checkbox toggles that batch-create or batch-delete permissions via the existing admin endpoints.
> 
> Updates `GET /v1/admin/permissions` to accept missing/empty `roles` (returning the full matrix), switch from raw SQL to `Permission.findAll`, include static report object IDs, and enforce `OBJECT_ID_PERMISSION_SCHEMA` so only allowed verbs appear/apply for object-level permissions.
> 
> Refactors permission constants/models to introduce `OBJECT_ID_PERMISSION_SCHEMA` (replacing `NOUNS_WITH_OBJECT_ID` as the source of truth), adds `VERB_DISPLAY_ORDER` for consistent UI summaries, and tightens `Permission.validatePermissionSchema` to validate objectId verbs/nouns more precisely; removes server tests expecting `422` for missing/empty `roles`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 086558fa8207b80719f863cdf179605f50ecc44d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/web/app/api/mutations/usePermissionMutation.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      const items = Array.isArray(toggles) ? toggles : [toggles];
+      const toCreate = [];
+      const toDelete = [];
+      for (const { verb, noun, objectId, roleId, hasPermission } of items) {
+        const params = { verb, noun, roleId, ...(objectId ? { objectId } : {}) };
+        if (hasPermission) {
+          toDelete.push(params);
+        } else {
+          toCreate.push(params);
+        }
```
[the remark this answers is not in the document]
Comment (chris-bes) [reply packages/web/app/api/mutations/usePermissionMutation.js]: I actually like hasPermission better

File: packages/web/app/views/administration/permissions/useNounOptions.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const seenKeys = new Set();
+    for (const perm of permissions) {
+      if (perm.objectId) {
+        objectIdGroupNouns.add(perm.noun);
+        const key = `${perm.noun}#${perm.objectId}`;
+        if (!seenKeys.has(key)) {
+          seenKeys.add(key);
+          const displayName = objectNames[key] ?? perm.objectId;
+          childEntries.push({
+            key,
```
[the remark this answers is not in the document]
Comment (chris-bes) [reply packages/web/app/views/administration/permissions/useNounOptions.js]: This is going to be used for Autocomplete Field. When a value is selected, the only thing being passed is the value. So we need to prefix 'objectId' so that the selected value can be differentiated between regular nouns and object IDs. It's a bit hacky but I don't think it's that bad

File: packages/web/app/views/administration/permissions/NounSection.jsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import { getVerbAbbreviation, usePermissionToggles } from './usePermissionToggles';
+
+export const CHEVRON_WIDTH = 32;
+
+export const stickyLeft = (left, bg = Colors.white) => `
+  position: sticky;
+  left: ${left}px;
+  background-color: ${bg};
+  z-index: 1;
+`;
```
Comment (jaskfla) [inline packages/web/app/views/administration/permissions/NounSection.jsx]: For formatting and autocomplete:

```suggestion
export const stickyLeft = (left, bg = Colors.white) => css`
  position: sticky;
  left: ${left}px;
  background-color: ${bg};
  z-index: 1;
`;
```

File: packages/web/app/views/administration/permissions/NounSection.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+  const objectName = nounGroup.objectId
+    ? objectNames[`${nounGroup.noun}#${nounGroup.objectId}`]
+    : null;
+
+  return (
+    <>
+      <NounRow onClick={() => setExpanded(prev => !prev)}>
+        <ChevronCell>
+          {expanded ? <KeyboardArrowUp fontSize="small" /> : <KeyboardArrowDown fontSize="small" />}
```
Comment (jaskfla) [inline packages/web/app/views/administration/permissions/NounSection.jsx]: I would challenge whether these are the correct directions for the arrows 🤔

The strong convention is to point → when collapsed and ↓ when expanded

Small UI enhancement: just use a single icon and pop a `transition` on `rotate` property

File: packages/web/app/views/administration/permissions/constants.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  // Noun-level permission (e.g. Patient, Encounter)
+  NOUN: 'noun',
+
+  // Expandable group containing per-object permission rows
+  // (e.g. Survey (objectID) which would expand to show all the individual Surveys)
+  OBJECT_ID_GROUP: 'objectIdGroup',
+
+  // Individual Child Object ID (e.g. Survey 123, Survey 456, Survey 789) within the group
+  OBJECT_ID: 'objectId',
+};
```
[the remark this answers is not in the document]
Comment (jaskfla) [reply packages/web/app/views/administration/permissions/constants.js]: Oh, sorry! In that case:

```js
const foo = /** @type {const} */ ({ foo: bar });
```

The `()` is important! And then this way when we “build” with `emitDeclarationOnly` we get the same output as if we did `as const` in a `.ts` file
